### PR TITLE
Include LICENSE.txt in the released jar files

### PR DESCRIPTION
**Pull request by brad4d**
I've created this as a pull request instead of a normal Google-internal change because it was easier to work on the OSS build rules from a git repository.

### CODE REVIEW

**Comment by niloc132** [comment]
I think the title might have a typo - if it is "unshaded", then it should _not_ include dependencies, by including dependencies, the jar then would be shaded. 

Of course there is also the distinction between "the shaded classes are rebased/repackaged" and just merging all of the jars, but it would hurt our use cases if the "unshaded" jars must then contain dependencies instead of relying on dependency management tools to look them up as needed.

File: BUILD.bazel [hunk 1]
```diff
@@ -108,21 +118,23 @@ java_export(
     name = "compiler_unshaded",
     maven_coordinates = "com.google.javascript:closure-compiler-unshaded:{0}".format(version),
     pom_template = "maven/closure-compiler-unshaded.pom.xml.tpl",
+    resources = [
+        ":LICENSE",
+        ":THIRD_PARTY_NOTICES",
+    ],
     runtime_deps = [
-        "//:compiler_unshaded_lib",
+        "//:compiler_unshaded_lib_deploy.jar",
```
[the remark this answers is not in the document]
Comment (brad4d) [reply BUILD.bazel:126]: Thank you for commenting.

I'll undo this change.

This was a desperate attempt to force the "LICENSE" file into the final jar that goes to Maven. The `java_export()` rule that is supposed to create the "unshaded" JAR absolutely refuses to include the `LICENSE` file, even though it quite happily includes the 'THIRD_PARTY_NOTICES' file.

I think there must be some special casing in the `java_export` rule implementation that explicitly skips any file named `LICENSE`. It's driving me nuts. I experimentally created a file named `GEORGE` included in exactly the same way as `LICENSE` and it got included in the final jar just fine.

If you have any advice for resolving this, I'd be delighted to hear it.

**Comment by brad4d** [comment]
I've just pushed a new version of this fix that I think is more satisfactory.

**Review by niloc132 — APPROVED** [review]
I can confirm that this seems to work as described, the resulting unshaded jar now has a LICENSE.txt file in its root.

I took a bit of time to go over `rules_jvm_external` (where `java_export` is defined) as well as `rules_java`, but neither seems to explicitly prevent LICENSE files from being copied. There are two odd issues in rules_jvm_external that could be related (https://github.com/bazelbuild/rules_jvm_external/issues/583, https://github.com/bazelbuild/rules_jvm_external/issues/572) but those seem to apply to an older version of rules_jvm_external than closure-compiler is presently using. I also tried updating to latest rules_java and rules_jvm_external, but that didn't avoid the issue.

One possible concern (or not), is that the shaded jar contains both LICENSE and LICENSE.txt - but appears to be the same contents. Do you want to exclude one or the other, or permit both?

**Comment by brad4d** [comment]
@gkdn fixed the underlying problem that this hacky PR was trying to fix, so it is no longer necessary.

**Comment by hjhasnah442-lgtm** [comment]
http://star-am.com

**Pull request closed without merging**